api/database/migrations/2025_02_14_145918_rename_dream_role.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Query\Expression;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('users', function (Blueprint $table) {
+
+            // 1. dream_role_classification_id becomes next_role_classification_id and career_objective_classification_id
+            $table->dropColumn('dream_role_classification_id');
+
+            $table->foreignUuid('next_role_classification_id')
+                ->nullable()
+                ->constrained(table: 'classifications', column: 'id')
+                ->onDelete('cascade');
+
+            $table->foreignUuid('career_objective_classification_id')
+                ->nullable()
+                ->constrained(table: 'classifications', column: 'id')
+                ->onDelete('cascade');
+
+            // 2. A pair of new fields: next_role_target_role and career_objective_target_role - a new enum TARGET_ROLE
+            $table->string('next_role_target_role')->nullable();
+            $table->string('career_objective_target_role')->nullable();
+
+            // 3. A pair of new fields to go with the enum above: next_role_target_role_other and career_objective_target_role_other - string
+            $table->string('next_role_target_role_other')->nullable();
+            $table->string('career_objective_target_role_other')->nullable();
+
+            // 4. dream_role_title becomes next_role_job_title and career_objective_job_title
+            $table->dropColumn('dream_role_title');
+            $table->string('next_role_job_title')->nullable();
+            $table->string('career_objective_job_title')->nullable();
+
+            // 5. dream_role_community_id becomes next_role_community_id and career_objective_community_id
+            $table->dropColumn('dream_role_community_id');
+
+            $table->foreignUuid('next_role_community_id')
+                ->nullable()
+                ->constrained(table: 'communities', column: 'id')
+                ->onDelete('cascade');
+
+            $table->foreignUuid('career_objective_community_id')
+                ->nullable()
+                ->constrained(table: 'communities', column: 'id')
+                ->onDelete('cascade');
```

Comment:
There appears to also be an "other community" state

![image](https://github.com/user-attachments/assets/7be11e61-977a-44a6-aa22-7f44aba5f42d)

Reply:
Oh shoot, I missed that.  That's going to be quite tricky to work into the data model because it's a real relationship, not an enum where we can just add an "OTHER" option to the set.  @JoshBeveridge could you please confirm if being able to choose an "other" community is in the requirements?  I know we dropped a few of these "other" choices.

Reply:
I think it should be straightforward 🤔 

Community id field (UUID) | community other name (string)
null, null = incomplete
id, null = existing community
null, string = other community option selected

Reply:
Will add to another issue

Reply:
Stubbed out https://github.com/GCTC-NTGC/gc-digital-talent/issues/12834